Cleanups.

christianhaeubl opened the pull request:
I removed all TODOs from the sources. We should probably collect the TODOs listed below in a separate issue:
- Rewrite some of the Java-level JFR events as Native events (e.g., JavaThreadStatistics)
- JfrChunkWriter.openFile should probably write all live threads
- JfrNativeEventWriter.accommodate can lose the current event if the new buffer is null
- JfrThreadLocal.beforeThreadRun - parentThread is null at the moment
- JfrThreadLocal.beforeThreadRun - write a thread checkpoint so that JFR knows about the newly started thread
- JfrTypeRepository - fix the 'exported' field
- TestSingleEvent - parse & verify the constant pool

christianhaeubl committed:
Reintroduced the JfrSerializerSupport.
Renamed JfrRepository to JfrConstantPool.
Removed all TODOs.

christianhaeubl committed:
Removed unused classes.

christianhaeubl committed:
Disallow Java synchronization in VM operations.

christianhaeubl committed:
Moved UninterruptibleHashTable.

christianhaeubl committed:
Moved all JFR test cases.

christianhaeubl committed:
Only enable JFR on JDK11.

christianhaeubl committed:
Minor style-related changes.

christianhaeubl committed:
Disable writing thread start/end events until we have more of the thread-related JFR infrastructure.

christianhaeubl commented:
Which command line do you currently use for executing the unit tests?

jiekang commented:
> Which command line do you currently use for executing the unit tests?

I think it's: `mx native-unittest com.oracle.svm.jfrtest --build-args -H:+AllowVMInspection`

jiekang commented on substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrGlobalMemory.java:
```diff
@@ -141,13 +142,13 @@ private static void releasePromotionBuffer(JfrBuffer buffer) {
-        // TODO: implement
-        return false;
+    private static void discardOldest() {
+        VMError.unimplemented();
```
Looks like `VMError.unimplemented()` is not uninterruptible, so maybe temporarily set callee to be false?

jiekang reviewed (changes requested):
One comment for `JfrGlobalMemory.shouldDiscard()`

There is also a failure on running atm with
```
No Java synchronization must be performed within a VMOperation: if the object is already locked, the VM is deadlocked
Fatal error: Should not reach here: Not okay to block.
```

Maybe after reworking on top of the chunk rotation that has the uninterruptible writes it'll be okay?

christianhaeubl committed:
Fixed issue regarding uninterruptible.

christianhaeubl committed:
Removed more unused code.

christianhaeubl commented:
Rebased and fixed the remaining issues. The unit tests pass now.
@rkennke : it seems weird that the method `JfrTraceIdLoadBarrier.clear()` is currently unused. Please have a brief look on that.

jiekang committed:
Merge branch 'jfr' into chaeubl/cleanups

rkennke commented:
> Rebased and fixed the remaining issues. The unit tests pass now.
> @rkennke : it seems weird that the method `JfrTraceIdLoadBarrier.clear()` is currently unused. Please have a brief look on that.

Yeah I guess it should be called right after epoch switch.
